report/sections/TestRunner/chapter.tex

```diff
@@ -1,33 +1,32 @@
 The GHC interpreter can be run from the command line with arguments specifying the path Haskell file to run.
-In other words, the Test Runner needs to write the user submission and corresponding test code to a file.
-However, if more users were to make a submission at the same time, the content of the file could potentially be overwritten while the GHC interpreter is running the tests.
-
+In other words, the Test Runner writes the user submission and corresponding test code to a file.
+However, if multiple users were to post a submission at the same time, the content of the file could potentially be overwritten while the GHC interpreter is running the tests.
 To address this, we generate a new directory with a unique name for each submission.
 Then, the user submission is written to two  \texttt{.hs} files inside this directory - a \texttt{code.hs} file containing the exercise submission, and a \texttt{test.hs} file containing the corresponding tests.
 This way, we can pass the path to this isolated directory to the GHC interpreter process as an argument.
 Once the tests have been run, the directory and its contents are deleted again.
 The \texttt{stdout} of the process is then read to determine whether the tests succeeded or not.
 Finally, the interpreter output and the test success flag is packaged into a JSON object and sent back to the client.
 
-Sequentially starting a new \texttt{runhaskell} process for each request was identified as a major bottleneck during testing, which will be described in detail in chapter \ref{chap:Benchmarking}.
-In order to address this, we started by refactoring the code base to allow running several instances of the GHC interpreter process at the same time.
-However, despite this running faster for a small number of clients, this approach in itself is not sufficiently stable:
-If multiple clients were to post a code submission at the same time, the Test Runner will start a new process for each request.
+Sequentially starting a new \texttt{runhaskell} process for each request is a major bottleneck when multiple users submit solutions in quick succession.
+Processing each request sequentially results in slow response times, which may also lead to connection timeouts.
+In order to address this,
+This was solved by allowing multiple instances of the GCH interpreter process to run at the same time.
+However, despite this being faster for a small number of clients, this approach in itself is not sufficient:
+If multiple clients post a code submission at the same time, the Test Runner will start a new process for each request.
 Given a large enough number of requests, this would crash the Test Runner web server.
 Therefore, we needed a way to limit the number of running GHC interpreter processes while at the same time storing subsequent requests for future processing.
 
+To solve the aforementioned problem, we decided to implement a queue system.
```

Comment:
```suggestion
To prevent crashes, we decided to implement a queue system.
```
In just a few lines, we mention two problems and make references to solutions. Might be better to specify which problem we're solving, like so?

Reply:
Maaske udvide saetningen. "implement a queue system enabling xxxx"

Reply:
```suggestion
To address the aforementioned problem where the system crashes under heavy use, we decided to implement a queue system.
```
Clarify that the system doesn't "just crash", but only does so under heavy use.